src/main/java/com/asu/cse545/group12/controller/TransactionController.java

```diff
@@ -43,22 +52,36 @@ public ModelAndView getCreditForm() {
 			logger.debug("credit Screen is executed!");
 		}
 		ModelAndView modelView = new ModelAndView();
-		modelView.addObject("form", new Form());
+		modelView.addObject("form", new TransactionForm());
 		modelView.setViewName("credit");
 		return modelView;
 	}
-	
-	@RequestMapping("creditAmount")
-	public ModelAndView creditAmount(@ModelAttribute("form") Form form, HttpServletRequest request) {
+
+	@RequestMapping(params = "creditAmount", method = RequestMethod.POST)
+	public ModelAndView creditAmount(@Valid @ModelAttribute("form") TransactionForm form, BindingResult result, HttpServletRequest request) {

```

Comment:
Good idea even I was thinking about adding bean to each form to do proper validation